activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb

```diff
@@ -373,6 +373,18 @@ def insert_fixtures(rows, table_name)
         end
       end
 
+      def insert_fixtures_set(fixture_set, tables_to_delete = [])
+        disable_referential_integrity do
+          transaction(requires_new: true) do
```

Comment:
Why is this `requires_new` but the other isn't?

Reply:
Good catch, my bad the other should have it too. It was extracted from here

https://github.com/rails/rails/blob/fd04c8cbb6627ef65ea728b5026b21bd24471c18/activerecord/lib/active_record/fixtures.rb#L558